File: src/transformers/features/protobuf/ProtobufCreateValidateEncodeTransformer.ts
import { ProtobufValidateEncodeProgrammer } from "../../../programmers/protobuf/ProtobufValidateEncodeProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateValidateEncodeTransformer {
    export const transform = GenericTransformer.factory(
        "protobuf.createValidateEncode",
    )(
        (project) => (modulo) =>
            ProtobufValidateEncodeProgrammer.write(project)(modulo),
    );
}

File: packages/transform/src/features/protobuf/ProtobufCreateValidateEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufCreateValidateEncodeTransformer = protobufCreateValidateEncodeTransformerNamespace{}

type protobufCreateValidateEncodeTransformerNamespace struct{}

func (protobufCreateValidateEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufValidateEncodeProgrammer.Write(call.TypeText)
}
